src/Common/src/Interop/Unix/System.Native/Interop.GetNativeIPInterfaceIndex.cs

```diff
@@ -0,0 +1,15 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetNativeIPInterfaceIndex")]
+        public static extern unsafe uint GetNativeIPInterfaceIndex(string name);
```

Comment:
In general we try to have the shims be as close to the native function as possible, so this should probably be named more like:
```C#
InterfaceNameToIndex
```
or something like that.